### PR TITLE
Add iterable support for IN criterias

### PR DESCRIPTION
**Pull request by l-vo**
Add support for iterables (including generators) for queries with IN/NOT IN criteria

### CODE REVIEW

**Comment by greg0ire** [comment]
Why are you targeting `master`?

**Comment by l-vo** [comment]
Because it's a new feature. I should. target another branch ?

**Comment by greg0ire** [comment]
Yes, 2.8.x is the branch for new features. `master` is for BC-breaks.

**Comment by l-vo** [comment]
Target branch updated. Thank you 🙂

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 1]
```diff
@@ -175,6 +176,39 @@ public function testCollectionParameters()
         $this->assertEquals($cities, $parameter->getValue());
     }
 
+    public function provideProcessParameterValueIterable() : Generator
+    {
+        $baseArray = [
+            0 => 'Paris',
+            3 => 'Canne',
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php:183]:
```suggestion
            3 => 'Cannes',
```

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php [hunk 2]
```diff
@@ -175,6 +176,39 @@ public function testCollectionParameters()
...
+    /**
+     * @dataProvider provideProcessParameterValueIterable
+     */
+    public function testProcessParameterValueIterable(iterable $cities) : void
+    {
+        $query = $this->_em->createQuery('SELECT a FROM Doctrine\Tests\Models\CMS\CmsAddress a WHERE a.city IN (:cities)');
+        self::assertEquals(
+            [
+                0 => 'Paris',
+                3 => 'Canne',
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Query/QueryTest.php:205]:
```suggestion
                3 => 'Cannes',
```

File: tests/Doctrine/Tests/ORM/Query/QueryTest.php (hunk 1 above)
Comment (l-vo) [reply tests/Doctrine/Tests/ORM/Query/QueryTest.php]: 😁 It was code that was already in the repository but I confess I didn't notice the type. Fixed :)

**Review by greg0ire — APPROVED** [review]

**Review by ostrolucky — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @l-vo !

**Comment by ambroisemaupate** [comment]
This introduces a BC on 2.8.x

https://github.com/doctrine/orm/issues/8365

Here is a test case : 

https://github.com/ambroisemaupate/orm/commit/eca3a3fb158e8d39de2b6db2d552d10c140544bd

**Comment by greg0ire** [comment]
@ambroisemaupate I think you mean a BC break ;)

**Pull request merged**